packages/adapter/addon/rest.js

```diff
@@ -1301,11 +1301,13 @@ function fetchSuccessHandler(adapter, payload, response, requestData) {
 
 function fetchErrorHandler(adapter, payload, response, errorThrown, requestData) {
   let responseData = fetchResponseData(response);
+
   if (responseData.status === 200 && payload instanceof Error) {
     responseData.errorThrown = payload;
     payload = responseData.errorThrown.payload;
   } else {
     responseData.errorThrown = errorThrown;
+    payload = adapter.parseErrorResponse(payload);
```

Comment:
A few comments after reading through the existing implementation:

I believe the necessary change is either this OR

```
  if (!response.ok) {
    return payload;
  }
```

^ this line in determineBodyPromise should possibly attempt JSON.parse. **I lean heavily towards the former (current PR) solution**

The primary difference is in whether we give the adapter the chance to do the parsing. Note the Ajax codepath calls this for all Ajax errors (I'm not sure if a 404 triggers an error though, as odd as that sounds at first glance, it doesn't trigger one in fetch for instance).

But if it does...

```
let payload = adapter.parseErrorResponse(jqXHR.responseText);
```

^ this is what would be doing the similar parsing for the Ajax branch.

cc @igorT